src/services/codefixes/fixUnusedIdentifier.ts

```diff
@@ -121,9 +121,20 @@ namespace ts.codefix {
                 break;
 
             case SyntaxKind.Parameter:
-                const functionDeclaration = <FunctionDeclaration>parent.parent;
-                if (functionDeclaration.parameters.length === 1) {
-                    changes.deleteNode(sourceFile, parent);
+                const oldFunction = parent.parent;
+                if (isArrowFunction(oldFunction) && oldFunction.parameters.length === 1) {
+                    const newFunction = updateArrowFunction(
+                        oldFunction,
+                        oldFunction.modifiers,
+                        oldFunction.typeParameters,
+                        /*parameters*/ undefined,
+                        oldFunction.type,
+                        oldFunction.equalsGreaterThanToken,
+                        oldFunction.body);
+
+                    suppressLeadingAndTrailingTrivia(newFunction);
+
+                    changes.replaceRange(sourceFile, { pos: oldFunction.getStart(), end: oldFunction.end }, newFunction);

```

Comment:
Could this use `replaceNode`?

Reply:
Hm, `replaceNode` starts with `node.getStart()` but then adjusts it -- maybe there's a bug in `getAdjustedStartPosition`, because it should be replacing a range that doesn't include leading trivia.

Reply:
I understood the adjusted start position was an attempt to establish the boundary between the trailing trivia of the preceding token and the leading trivia of the current token.

Reply:
Well, that's a problem if we want to preserve the leading trivia of the current token -- maybe all the uses of `replaceNode` have this bug and `replaceNode` should be fixed to just use `getStart()`.

Reply:
Yes, every caller of `replaceNode` probably needs to be reviewed.  Few of the existing tests include interesting trivia.

Reply:
OK, I've created an issue for this at #21246. This PR should be good to go for now.